guides/source/active_support_instrumentation.md

```diff
@@ -3,623 +3,1145 @@
+[Active Support](https://api.rubyonrails.org/classes/ActiveSupport.html) is a
+core component of Ruby on Rails, offering a wide range of Ruby language
+extensions and utility classes.
 
-In this guide, you will learn how to use the Active Support's instrumentation API to measure events inside of Rails and other Ruby code.
+One of its key features is the Instrumentation API, which is described in
+further detail in this guide. You will learn:
 
-After reading this guide, you will know:
-
-* What instrumentation can provide.
-* How to add a subscriber to a hook.
-* The hooks inside the Rails framework for instrumentation.
-* How to build a custom instrumentation implementation.
+* About Instrumentation
+* How to instrument your own events.
+* How to subscribe to events.
+* The built-in instrumentation events available inside Rails.
 
 --------------------------------------------------------------------------------
 
 Introduction to Instrumentation
 -------------------------------
 
-The instrumentation API provided by Active Support allows developers to provide hooks which other developers may hook into. There are [several of these](#rails-framework-hooks) within the Rails framework. With this API, developers can choose to be notified when certain events occur inside their application or another piece of Ruby code.
+The Instrumentation API provides a way to instrument code and subscribe to
+events that occur in your application. Instrumentation means wrapping a block of
+code so that, when it runs, an event is emitted with a name and optional
+payload. Any subscribers listening for that event will then be notified and can
+react, for example, they can log information, benchmark, or perform some other
+action. This makes it possible to observe behavior within the Rails framework,
+in your own code, or even in standalone Ruby scripts. There are a few parts that
+are vital to understanding the Instrumentation API: events and subscribers.
 
-For example, there is [a hook](#sql-active-record) provided within Active Record that is called every time Active Record uses an SQL query on a database. This hook could be **subscribed** to, and used to track the number of queries during a certain action. There's [another hook](#process-action-action-controller) around the processing of an action of a controller. This could be used, for instance, to track how long a specific action has taken.
+NOTE: The Instrumentation API is intended for framework code, not application
+code. For application-level event reporting, use [Active Support Structured
+Events](active_support_structured_events.html) instead.
```

Comment:
NOTE: this does not exist as yet, and I will keep an eye on this [PR](https://github.com/rails/rails/pull/56156/files#diff-092425391b603bb5f9f734a92795cef68893151e6834edba7796b0ecec6439af) before merging this one.